common/src/main/java/com/microsoft/identity/common/internal/util/DateUtilities.java

```diff
@@ -22,11 +22,55 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.util;
 
+import androidx.annotation.NonNull;
+
 import java.util.Date;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
 public final class DateUtilities {
 
+    /**
+     * Static lock object to use when performing operations that modify locale state of JVM.
+     */
+    public static final Object LOCALE_CHANGE_LOCK = new Object();
+
+    private static final String LOCALE_PREFIX_ARABIC = "ar";
+    private static final String LOCALE_PREFIX_ASSAMESE = "as";
+    private static final String LOCALE_PREFIX_BENGALI = "bn";
+    private static final String LOCALE_PREFIX_ALGERIAN = "dz";
+    private static final String LOCALE_PREFIX_PERSIAN = "fa";
+    private static final String LOCALE_PREFIX_KASHMIRI = "ks";
+    private static final String LOCALE_PREFIX_MARATHI = "mr";
+    private static final String LOCALE_PREFIX_BURMESE = "my";
+    private static final String LOCALE_PREFIX_NEPALI = "ne";
+    private static final String LOCALE_PREFIX_PUNJABI = "pa";
+    private static final String LOCALE_PREFIX_PASHTO = "ps";
+    private static final String LOCALE_PREFIX_URDU = "ur";
+    private static final String LOCALE_PREFIX_UZBEK = "uz";
+
+    // This list may not be exhaustive, but represents the set of non-Gregorian locales
+    // available as of AOSP API 24
+    private static final Set<String> NON_GREGORIAN_CALENDAR_LOCALES = new HashSet<>();
```

Comment:
You can initialize your hash set with a Collection using Arrays.asList().

Reply:
e65bcf2ca599fd50d835c03d97d8f0545d23d3ee